.github/workflows/benchmark.yml

```diff
@@ -24,71 +24,111 @@ jobs:
     name: Performance regression check
     runs-on: ubuntu-latest
     steps:
-      - name: Set benchmark repetitions
-        # reducing repetition will speed up execution,
-        # but will be more inaccurate at detecting change
-        run: echo "benchmark_repetitions=7" >> "$GITHUB_OUTPUT"
-        id: settings
-
-      - name: Install dependencies
-        run: sudo apt-get update && sudo apt-get install wabt
 
       - name: Checkout
         uses: actions/checkout@v3
         with:
           fetch-depth: 0
 
+      - name: Detect changed components
+        uses: dorny/paths-filter@v3
+        id: changes
+        with:
+          filters: |
+            encoding:
+              - 'encoding/**'
+            parser:
+              - 'parser/**'
+            sema:
+              - 'sema/**'
+            bbq:
+              - 'bbq/**'
+
+      - name: Set benchmark settings
+        # reducing repetition and time will speed up execution,
+        # but will be more inaccurate at detecting change
+        run: |
+          echo "benchmark_repetitions=2" >> "$GITHUB_OUTPUT"
+          echo "benchmark_time=2s" >> "$GITHUB_OUTPUT"
+
+        id: settings
+
       - uses: actions/setup-go@v3
         with:
           go-version: ${{ env.GO_VERSION }}
           cache: true
 
-      - name: Build
-        run: make -j8 build
+      - name: Run unconditional benchmarks, on current branch
+        run: make bench-common BENCH_REPS=${{ steps.settings.outputs.benchmark_repetitions }} BENCH_TIME=${{ steps.settings.outputs.benchmark_time }} | tee new.txt
 
-      - name: Run benchmark on current branch
-        shell: bash
-        run: |
-          ( for i in {1..${{ steps.settings.outputs.benchmark_repetitions }}}; do go test ./... -run=XXX -bench=. -benchmem -shuffle=on; done | sed 's/pkg:.*/pkg: github.com\/onflow\/cadence\//' ) | tee new.txt
-      # the package replace line above is to make the results table more readable, since it is not fragmented by package
+      - name: Run encoding benchmarks, on current branch
```

Comment:
It would be nice to find a way to iterate over `steps.changes.outputs` and collapse this into one step